Analysis/Core/include/AnalysisCore/TrackSelectorPID.h

```diff
@@ -0,0 +1,279 @@
+  /// \param track  track
+  /// \param conditionalTPC  variable to store the result of selection with looser cuts for conditional accepting of track if combined with TPC
+  /// \return true if track satisfies TOF PID hypothesis for given TOF nσ range
+  template <typename T>
+  bool isSelectedTrackPIDTOF(const T& track, bool& conditionalTPC)
+  {
+    // Accept if selection is disabled via large values.
+    if (mNSigmaTOFMin < -999. && mNSigmaTOFMax > 999.) {
+      return true;
+    }
+
+    // Get nσ for a given particle hypothesis.
+    double nSigma = 100.;
+    switch (mPdg) {
+      case kPiPlus: {
+        nSigma = track.tofNSigmaPi();
+        break;
+      }
+      case kKPlus: {
+        nSigma = track.tofNSigmaKa();
+        break;
+      }
+      case kProton: {
+        nSigma = track.tofNSigmaPr();
+        break;
+      }
+      default: {
+        LOGF(error, "ERROR: TOF PID not implemented for PDG %d", mPdg);
+        assert(false);
+      }
+    }
+
+    if (mNSigmaTOFMinCondTPC < -999. && mNSigmaTOFMaxCondTPC > 999.) {
+      conditionalTPC = true;
+    } else {
+      conditionalTPC = mNSigmaTOFMinCondTPC <= nSigma && nSigma <= mNSigmaTOFMaxCondTPC;
+    }
+    return mNSigmaTOFMin <= nSigma && nSigma <= mNSigmaTOFMax;
+  }
+
+  /// Returns status of TOF PID selection for a given track.
+  /// \param track  track
+  /// \return TOF selection status (see TrackSelectorPID::Status)
+  template <typename T>
+  int getStatusTrackPIDTOF(const T& track)
+  {
+    if (isValidTrackPIDTOF(track)) {
+      bool condTPC = false;
+      if (isSelectedTrackPIDTOF(track, condTPC)) {
+        return Status::PIDAccepted; // accepted
+      } else if (condTPC) {
+        return Status::PIDConditional; // potential to be accepted if combined with TPC
+      } else {
+        return Status::PIDRejected; // rejected
+      }
+    } else {
+      return Status::PIDNotApplicable; // PID not applicable
+    }
+  }
+
+  /// Returns status of combined PID selection for a given track.
+  /// \param track  track
+  /// \return combined-selection status (see TrackSelectorPID::Status)
+  template <typename T>
+  int getStatusTrackPIDAll(const T& track)
+  {
+    int statusTPC = getStatusTrackPIDTPC(track);
+    int statusTOF = getStatusTrackPIDTOF(track);
+
+    if (statusTPC == Status::PIDAccepted || statusTOF == Status::PIDAccepted) {
+      return Status::PIDAccepted; // what if we have Accepted for one and Rejected for the other?
+    }
+    if (statusTPC == Status::PIDConditional && statusTOF == Status::PIDConditional) {
+      return Status::PIDAccepted;
+    }
+    if (statusTPC == Status::PIDRejected || statusTOF == Status::PIDRejected) {
+      return Status::PIDRejected;
+    }
+    return Status::PIDNotApplicable; // (NotApplicable for one detector) and (NotApplicable or Conditional for the other)
+  }
+
+ private:
+  uint mPdg = kPiPlus; ///< PDG code of the expected particle
```

Comment:
Why not templated?

Reply:
Let's do it as a next iteration. I think the priority now is to get rid of the duplicate code.